worker.go

```diff
@@ -31,8 +32,14 @@ import (
 type (
 	// Worker represents objects that can be started and stopped.
 	Worker interface {
-		Stop()
+		// Start starts the worker in a non-blocking fashion
 		Start() error
+		// Run is a blocking start and cleans up resources when killed
+		Run()
```

Comment:
Agree that minimal options are better in general. Here we need:
1. a blocking start method
2. a non blocking start method that returns once the worker is started

The challenge with starting in a goroutine is the user has no way of knowing if the worker even started/errored out etc. So they will proceed doing work when the worker might actually not have started at all.